package/kedro_viz/server.py

```diff
@@ -474,7 +489,39 @@ def format_pipeline_data(
 @app.route("/api/nodes.json")
 def nodes_json():
     """Serve the pipeline data."""
-    return jsonify(data)
+    return jsonify(_DATA)
+
+
+@app.route("/api/nodes/<string:node_id>")
+def nodes_metadata(node_id):
+    """Serve the metadata for node and dataset."""
+    node = _NODES.get(node_id)
+    if node and node.get("type") == "task":
```

Comment:
Why do we need `node.get("type")`? Aren't `type` guaranteed to be in there?

Reply:
Yes, I can change it to `node["type"] == "task"`:)